config/app/api/openapi/registry.yml

```
@@ -0,0 +1,117 @@
+    license:
+        name: MIT
+paths:
+    '/private/application-config':
+        post: &paths/private/application-config/post
+            summary: 'Saves or updates application configuration between Tenant and App.'
+            parameters:
+                -   name: X-Store-Reference
+                    in: header
+                    required: true
+                    description: 'Reference of the Tenants store.'
+                    schema:
+                        type: string
+            requestBody:
+                description: 'Provides configuration for the application.'
+                required: true
+                content:
+                    application/json:
+                        schema:
+                            $ref: '#/components/schemas/ConfigurationApiRequest'
+            responses:
+                200:
+                    description: 'Expected response to a valid request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ConfigurationApiResponse'
+                400:
+                    description: 'Bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+                default:
+                    description: 'Expected response to a bad request.'
+                    content:
+                        text/plain:
+                            schema:
+                                type: string
+        put: *paths/private/application-config/post # $ref is not (officially) supported for operation objects
+
+    '/private/application/disconnect':
+        post:
+            summary: 'Disconnects Application from a Tenants store.
+                      Finds Application configuration and removes it.'
+            parameters:
+                -   name: X-Store-Reference
+                    in: header
+                    required: true
+                    description: 'Reference of the Tenants store.'
+                    schema:
+                        type: string
+            responses:
+                204:
+                    description: 'App is disconnected.'
+                400:
+                    description: 'Bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+                default:
+                    description: 'Expected response to a bad request.'
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ApiErrorMessages'
+components:
+    schemas:
+        ConfigurationApiResponse:
+            type: object
+            additionalProperties: true
```

```suggestion
        ConfigurationApiResponse:
            type: object
            additionalProperties: true
            example:
                configKey1: "config value"
                configKey2: ["selectedValue1", "selectedValue2"]
```
Same as the request example